### PR TITLE
Tactile steps

### LINKED ISSUE #3534: [Quest] Do these steps have tactile paving at the top and bottom?
Apologies for mistakes, this is my first attempt at this:

### General
Affected tag(s) to be modified/added: [tactile_paving=] (https://wiki.openstreetmap.org/wiki/Key:tactile_paving)
Question asked: Do these steps have tactile paving at the top and bottom?
Answers same as crossing Yes / No / Can't Say

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete)):
- [ ] 🚧 To be added tag is established and has a useful purpose - highway=steps has 16 913 instances of tactile_paving
- [ ] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one) - yes
- [ ] 🐿️ Easily answerable by everyone from the outside but a survey is necessary - yes
- [ ] 💤 Not an overwhelming percentage of quests have the same answer (No spam) - Most are no, but more and more are yes as areas are redeveloped
- [ ] 🕓 Applies to a reasonable number of map data (Worth the effort) - believe so. I am in the UK. Not sure about other countries?

### Ideas for implementation
Copy and paste the crossing quest :-)

highway=steps
tactile_paving is unset

**Metadata needed:**  No meta data needed, but probably enabled/disabled on country by country basis

<!-- If you have any idea on how the UI (the form appearing when the quest is asked) should look like.
I am willing to try and get some photos, if needed, but the same images for Crossing can be used.

### PR DESCRIPTION
**Pull request by arrival-spring**
Fixes #3534 now that it is not marked as blocked, follows on from work in #3762

Again, my first attempt at an icon (combining tactile paving and steps icons)

There are three new strings here as the current string for incorrect tactile paving is "Only on one side" and this doesn't make sense for steps, so it needed at least a string "Only at one end" and to me it makes sense to have separate options for top and bottom.

Tested and working and tests all pass

### CODE REVIEW

File: res/graphics/quest/steps_tactile_paving.svg
Comment (westnordost) [inline res/graphics/quest/steps_tactile_paving.svg:1]: The right end of the steps are a bit too close to the edge.

In Inkscape I usually add a grid (cell size 4x4) and try to align the drawings to that, for better visibility when scaled down.

File: app/src/main/java/de/westnordost/streetcomplete/quests/tactile_paving/AddTactilePavingSteps.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.osm.surface.ANYTHING_PAVED
+import de.westnordost.streetcomplete.osm.updateWithCheckDate
+
+class AddTactilePavingSteps : OsmFilterQuestType<TactilePavingStepsAnswer>() {
+
+    override val elementFilter = """
+        ways with highway = steps
+         and surface ~ ${ANYTHING_PAVED.joinToString("|")}
+         and (!indoor or indoor = no)
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/tactile_paving/AddTactilePavingSteps.kt:16]: Hmm, why not indoor, though? Maybe rather not conveying?

File: app/src/main/java/de/westnordost/streetcomplete/quests/tactile_paving/AddTactilePavingSteps.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+         and access !~ private|no
+        and (
+          !tactile_paving
+          or tactile_paving = unknown
+          or tactile_paving = no and tactile_paving older today -4 years
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/tactile_paving/AddTactilePavingSteps.kt:21]:
```suggestion
          or tactile_paving ~ no|partial|incorrect and tactile_paving older today -4 years
```
?

**Review by westnordost — COMMENTED** [review]
Thank you, looks good. Just a few remarks

File: res/graphics/quest/steps_tactile_paving.svg
Comment (arrival-spring) [reply res/graphics/quest/steps_tactile_paving.svg:1]: I've adjust it, how's that?

**Review by westnordost — APPROVED** [review]

File: res/graphics/quest/steps_tactile_paving.svg
Comment (westnordost) [reply res/graphics/quest/steps_tactile_paving.svg:1]: It's better, thanks

**Pull request merged**